### PR TITLE
Optimize dexmaker for yahfa

### PR DESCRIPTION
**Pull request by yujincheng08**
By moving things to a predefined class, dexmaker could run faster because it generates less code.

*Before*
```
D/EdXposed: generated class for public static java.lang.Integer de.robv.android.xposed.installer.XposedApp.getActiveXposedVersion() in 41.673077ms
```
*After Optimized*

```
D/EdXposed: generated class for public static java.lang.Integer de.robv.android.xposed.installer.XposedApp.getActiveXposedVersion() in 21.121720999999997ms
```
*After Cached*
```
D/EdXposed: generated class for public static java.lang.Integer de.robv.android.xposed.installer.XposedApp.getActiveXposedVersion() in 4.651198ms
```

Other TODOs (won't do in this PR):
1. Same trick can be applied to SandHook, see `edxp-sandhook/src/main/java/com/swift/sandhook/xposedcompat/methodgen/HookerDexMaker.java`

### CODE REVIEW

File: edxp-core/src/main/cpp/main/src/config_manager.h [hunk 1]
```diff
@@ -68,6 +68,10 @@ namespace edxp {
             return misc_path_ / "framework" / suffix;
         }
 
+        inline static auto GetCachePath(const std::string &suffix = {}) {
+            return misc_path_ / "cache" / suffix;
```
Comment (kotori2) [inline edxp-core/src/main/cpp/main/src/config_manager.h:72]: does `misc_path_` includes `edxp_xxx` prefix?

Comment (kotori2) [reply edxp-core/src/main/cpp/main/src/config_manager.h:72]: Also I think it's better to split cache for YAHFA and SandHook

Comment (yujincheng08) [reply edxp-core/src/main/cpp/main/src/config_manager.h:72]: > does `misc_path_` includes `edxp_xxx` prefix?

Ofcourse. See how it loads.

Comment (yujincheng08) [reply edxp-core/src/main/cpp/main/src/config_manager.h:72]: > Also I think it's better to split cache for YAHFA and SandHook

They have different prefixes.

**Pull request closed without merging**